main/lsp/requests/hover.cc

```diff
@@ -24,11 +24,38 @@ string methodSignatureString(const core::GlobalState &gs, const core::TypePtr &r
     return contents;
 }
 
-unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string str) {
-    if (markupKind == MarkupKind::Markdown && str.length() > 0) {
-        str = fmt::format("```ruby\n{}\n```", str);
+unique_ptr<MarkupContent> formatRubyCode(MarkupKind markupKind, string_view sigString,
+                                         optional<string_view> optionalDocString) {
+    string_view docString = optionalDocString.value_or("");
+    // Get rid of at most 1 trailing newline
+    if (docString.length() > 0 && docString.back() == '\n') {
+        docString.remove_suffix(1);
     }
-    return make_unique<MarkupContent>(markupKind, move(str));
+    if (sigString.length() > 0 && sigString.back() == '\n') {
```

Comment:
Ditto for `sigString`, except the trimming should be in `methodSignatureString` (and can use the `StripTrailingAsciiWhitespace` helper).

Reply:
74e0d5ada